pkg/process/checks/process_events_consumer.go

```diff
@@ -0,0 +1,163 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package checks
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/eventmonitor"
+	"github.com/DataDog/datadog-agent/pkg/eventmonitor/proto/api"
+	"github.com/DataDog/datadog-agent/pkg/process/events/model"
+	"github.com/DataDog/datadog-agent/pkg/security/metrics"
+	smodel "github.com/DataDog/datadog-agent/pkg/security/secl/model"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-go/v5/statsd"
+	"go.uber.org/atomic"
+	"time"
+)
+
+// ProcessConsumer receives events from the event monitoring module of the system-probe,
+// batches them in the messages channel and serves the messages over GRPC when requested
+type ProcessConsumer struct {
```

Comment:
What do you think of leaving a comment here explicitly saying that this consumer runs in the `system-probe` process just to prevent confusion with the listener in `process-agent`?